src/shared/utilities/timeoutUtils.ts

```diff
@@ -18,6 +19,19 @@ export class TimeoutError extends Error {
     public static isCancelled(err: any): err is TimeoutError & { type: 'cancelled' } {
         return err instanceof TimeoutError && err.type === 'cancelled'
     }
+
+    public static isExpired(err: any): err is TimeoutError & { type: 'expired' } {
+        return err instanceof TimeoutError && err.type === 'expired'
+    }
+}
+
+interface CancelEvent {
+    readonly type: 'expired' | 'cancelled'
+}
+
+/** A {@link CancellationToken} that provides a reasoning for the cancellation event. */
+interface TypedCancellationToken extends CancellationToken {
```

Comment:
This is to help ease into using more of VS Code's `CancellationToken`

Reply:
Can you clarify that statement: How does this ease us into the concept? I'm cool with just using CancellationTokens instead of an interface that we might remove later.

Reply:
`CancellationToken` is already an interface, but they're usually provided by VS Code APIs directly. But if we start using the same interface for things that aren't directly tied to VS Code it will keep things more uniform. The interfaces are simple enough that I think it makes sense.